servers/src/main/java/tachyon/master/journal/JournalWriter.java

```diff
@@ -328,8 +328,14 @@ public synchronized void flush() throws IOException {
       if (mOutputStream instanceof FSDataOutputStream) {
         ((FSDataOutputStream) mOutputStream).sync();
       }
-      if (mOutputStream.size() > mMaxLogSize) {
-        LOG.info("Rotating log file. size: " + mOutputStream.size() + " maxSize: " + mMaxLogSize);
+      boolean overSize = mOutputStream.size() > mMaxLogSize;
+      if (overSize || mUfs.getUnderFSType() == UnderFileSystem.UnderFSType.S3) {
```

Comment:
I feel safer to determine in this way which type the underfs is instead of determining from underfs address like `s3://`, or `s3n://`.